rosidl_generator_rs/rosidl_generator_rs/__init__.py

```diff
@@ -198,107 +223,88 @@ def constant_value_to_rs(type_, value):
     assert False, "unknown constant type '%s'" % type_
 
 
-def get_builtin_rs_type(type_):
-    if type_ == 'bool':
-        return 'bool'
-
-    if type_ == 'byte':
-        return 'u8'
-
-    if type_ == 'char':
-        return 'char'
-
-    if type_ == 'float32':
-        return 'f32'
-
-    if type_ == 'float64':
-        return 'f64'
-
-    if type_ == 'int8':
-        return 'i8'
-
-    if type_ == 'uint8':
-        return 'u8'
-
-    if type_ == 'int16':
-        return 'i16'
-
-    if type_ == 'uint16':
-        return 'u16'
-
-    if type_ == 'int32':
-        return 'i32'
-
-    if type_ == 'uint32':
-        return 'u32'
-
-    if type_ == 'int64':
-        return 'i64'
-
-    if type_ == 'uint64':
-        return 'u64'
-
-    if type_ == 'string':
+def get_builtin_rs_type(type_, package_name=None):
+    if isinstance(type_, BasicType):
+        if type_.typename == 'boolean':
+            return 'bool'
+        elif type_.typename in ['byte', 'octet']:
+            return 'u8'
+        elif type_.typename == 'char':
+            return 'u8'
+        elif type_.typename in ['float']:
+            return 'f32'
+        elif type_.typename in ['double', 'long double']:
+            return 'f64'
+        elif type_.typename == 'int8':
+            return 'i8'
+        elif type_.typename == 'uint8':
+            return 'u8'
+        elif type_.typename == 'int16':
+            return 'i16'
+        elif type_.typename == 'uint16':
+            return 'u16'
+        elif type_.typename == 'int32':
+            return 'i32'
+        elif type_.typename == 'uint32':
+            return 'u32'
+        elif type_.typename == 'int64':
+            return 'i64'
+        elif type_.typename == 'uint64':
+            return 'u64'
+    elif isinstance(type_, AbstractGenericString):
         return 'std::string::String'
+    elif isinstance(type_, Array):
+        return '[{}, {}]'.format(get_rs_type(type_.value_type), 32 if type_.size <= 32 else 32)
```

Comment:
I think this syntax for [defining Rust arrays of a fixed size ](https://doc.rust-lang.org/std/primitive.array.html)is slightly wrong. Instead of `[type, length]` shouldn't it be `[type; length]`? 
I'm a Rust n00b, so could be wrong, but I've added a dependency on `sensor_msgs` (which depends on `geometry_msgs`) in my project and got 3 errors like:

```
   Compiling geometry_msgs v0.1.0 (/home/loy/ros2_rust_ws/install/geometry_msgs/share/geometry_msgs/rust)
error: expected one of `!`, `(`, `+`, `::`, `;`, `<`, or `]`, found `,`
   --> /home/loy/ros2_rust_ws/install/geometry_msgs/share/geometry_msgs/rust/src/msg.rs:150:25
    |
150 |     pub covariance: [f64, 32],
    |                         ^ expected one of 7 possible tokens
```

With the change above, no such errors. One other remained, but that's a [separate issue](https://github.com/ros2-rust/ros2_rust/issues/26).

Reply:
Ah crap. Looks like you're right. I guess these packages don't get compiled unless you are importing `geometry_msgs`, since when I run `colcon build` I get:

```
Starting >>> geometry_msgs
Finished <<< geometry_msgs [2.91s]
```

We'll probably need ROS tests to cover all the standard ROS messages.

Reply:
Sometimes I hate being right :-). I'll make a PR